static/src/javascripts/projects/common/modules/atoms/youtube-player.ts

```diff
@@ -154,37 +152,30 @@ const onPlayerReadyEvent = (
 		}
 	});
 
-	// we should be able to remove this check once everything is using flow/ES^
-	if (handlers && typeof handlers.onPlayerReady === 'function') {
-		handlers.onPlayerReady(event);
-	}
+	handlers?.onPlayerReady(event);
+};
+
+interface AdFreeConfig {
+	disableAds: true;
+}
+const createAdFreeConfig = (): AdFreeConfig => {
+	const adFreeConfig: AdFreeConfig = { disableAds: true };
+	log('commercial', 'YouTube Ad-Free Config', adFreeConfig);
+	return adFreeConfig;
 };
```

Comment:
@arelra I know you were fighting with these types, but here I’ve just chosen to split them out entirely.

The two configs are completely independent, and this way we can forgo checks on state for ad-free.

Reply:
Yeah I ended up adding very similar types to `commercial-core` for DCR and `YoutubeAtom`.

https://github.com/guardian/commercial-core/blob/main/src/types.ts#L72

Not necessary for now but perhaps you could reuse them?

Reply:
Definitely worth importing here! Will add some more commits!